### PR TITLE
hotfix: remove empty decisions from document on save

### PR DESCRIPTION
**Pull request by usrtim**
In order to prevent users from being able to create “faulty” documents by copy pasting etc, we would remove decisions that don’t have any content from the document when saving.

### CODE REVIEW

File: app/controllers/agendapoints/edit.js [hunk 1]
```diff
@@ -93,14 +93,32 @@ export default class AgendapointsEditController extends Controller {
...
     }
   }
+
+  removeEmptyDivs(html) {
+    const parserInstance = new DOMParser();
+    const parsedHtml = parserInstance.parseFromString(html, 'text/html');
+    const body = parsedHtml.body.childNodes;
+
+    for (const child of body) {
+      if (child.attributes?.typeof?.textContent.includes('besluit:Besluit')) {
```
Comment (nvdk) [inline app/controllers/agendapoints/edit.js:115]: I would check for both the prefixed and full URI here.

File: app/controllers/agendapoints/edit.js [hunk 2]
```diff
@@ -93,14 +93,32 @@ export default class AgendapointsEditController extends Controller {
...
+        if (child.textContent.trim() === '') {
+          child.remove();
+        }
+      }
+    }
+
+    return parsedHtml.documentElement.lastChild.innerHTML;
```
Comment (nvdk) [inline app/controllers/agendapoints/edit.js:122]: shouldn't this just be `parsedHtml.body.innerHTML;` ?  That's more inline with how you start above

Comment (usrtim) [reply app/controllers/agendapoints/edit.js:122]: yes, thanks for noticing that.

File: app/controllers/agendapoints/edit.js (hunk 1 above)
Comment (usrtim) [reply app/controllers/agendapoints/edit.js]: @nvdk please let me know if new commit address your concern ?

**Review by elpoelma — APPROVED** [review]
Works as expected. When removing all content of a decision (but keeping its div) it is correctly removed.

**Review by abeforgit — CHANGES_REQUESTED** [review]
This only checks the top level children of the document. What if the div in question is further down the tree?
As we mentioned in the standup, there is https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll which will probably be useful here

**Pull request merged**